projects/packages/publicize/src/rest-api/class-share-status-controller.php

```diff
@@ -0,0 +1,186 @@
+			}
+
+			$done = metadata_exists( 'post', $post_id, REST_CONTROLLER::SOCIAL_SHARES_POST_META_KEY );
+
+			$response = array(
+				'shares' => $done ? $shares : array(),
+				'done'   => $done,
+			);
+
+			return rest_ensure_response( $response );
+		}
+
+		$response = $this->proxy_request_to_wpcom_as_user( $request );
+
+		if ( is_wp_error( $response ) ) {
+			return rest_ensure_response( $response );
+		}
+
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Verify that the request has access to Jetpack Social data.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error
+	 */
+	public function get_items_permissions_check( $request ) {// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		return $this->publicize_permissions_check();
+	}
+
+	/**
+	 * Schema for the endpoint.
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'jetpack-social-share-status',
+			'type'       => 'object',
+			'properties' => array(
+				'shares' => array(
+					'description' => __( 'List of shares.', 'jetpack-publicize-pkg' ),
+					'type'        => 'array',
+					'items'       => array(
+						'type'       => 'object',
+						'properties' => array(
+							'status'          => array(
+								'description' => __( 'Status of the share.', 'jetpack-publicize-pkg' ),
+								'type'        => 'string',
+							),
+							'message'         => array(
+								'description' => __( 'Share message or link.', 'jetpack-publicize-pkg' ),
+								'type'        => 'string',
+							),
+							'timestamp'       => array(
+								'description' => __( 'Timestamp of the share.', 'jetpack-publicize-pkg' ),
+								'type'        => 'integer',
+							),
+							'service'         => array(
+								'description' => __( 'The service to which it was shared.', 'jetpack-publicize-pkg' ),
+								'type'        => 'string',
+							),
+							'connection_id'   => array(
+								'description' => __( 'Connection ID for the share.', 'jetpack-publicize-pkg' ),
+								'type'        => 'integer',
+							),
+							'external_id'     => array(
+								'description' => __( 'External ID of the shared post.', 'jetpack-publicize-pkg' ),
+								'type'        => 'string',
+							),
+							'external_name'   => array(
+								'description' => __( 'External name of the shared post.', 'jetpack-publicize-pkg' ),
+								'type'        => 'string',
+							),
+							'profile_picture' => array(
+								'description' => __( 'Profile picture URL of the account sharing.', 'jetpack-publicize-pkg' ),
+								'type'        => 'string',
+							),
+							'profile_link'    => array(
+								'description' => __( 'Profile link of the sharing account.', 'jetpack-publicize-pkg' ),
+								'type'        => 'string',
+							),
+						),
+					),
+				),
+				'done'   => array(
```

Comment:
To make this endpoint compatible with `getEntityRecord` API, we may need to move this `done` flag to response headers and directly have that `shares` schema one level up.

Reply:
Yeah I thought in this PR we can follow the current implementation and in the next change we chan switch to using `getEntityRecord` to keep it more modular